pkg/rancher/migrations.go

```diff
@@ -132,6 +141,33 @@ func createOrUpdateConfigMap(configMapClient controllerv1.ConfigMapClient, cm *v
 	return err
 }
 
+func versionTombstone(configMapController controllerv1.ConfigMapController) error {
+	cm, err := getConfigMap(configMapController, rancherVersionTombstoneConfig)
+	if err != nil || cm == nil {
+		return err
+	}
+
+	if !semver.IsValid(rancherversion.Version) {
+		logrus.Errorf("rancher version %s is not semver compliant, skipping tombstone validation", rancherversion.Version)
+		return nil
+	}
+
+	lastVersion := cm.Data[rancherVersionKey]
+	if lastVersion != "" {
+		if !semver.IsValid(lastVersion) {
+			logrus.Errorf("Previous Rancher version %s is not semver compliant, skipping tombstone validation", lastVersion)
+			lastVersion = ""
+		}
+		if semver.Compare(lastVersion, rancherversion.Version) == 1 {
+			logrus.Fatalf("Detected Rancher downgrade from %s to %s. In order to perform a rollback of Rancher, use the Rancher Backup Restore Operator. If a suitable backup is unavailable, version tombstone validation can be temporarily disabled by deleting the rancherVersionTombstone config map in the cattle-system namespace.",
```

Comment:
I don't think that we should be using `Fatalf` here - `Fatalf` will cause the program to crash. Since we can have other threads running which may need to clean up work this could cause wider problems for the system. I think that this should return an error, and another part of the application (such as the Rancher Start() function) can choose if to crash due to this, and how it should crash.

Reply:
I'm open to suggestions on where would be a better spot to put this, since `OnLeader` isn't a great spot to put this anyway as there is no way to consume the error since `OnLeader` will just be run again.